kernels/portable/cpu/util/reduce_util.h

```diff
@@ -663,5 +664,23 @@ bool check_prod_out_args(
 
 #endif
 
+/**
+ * parallel_for wrapper for reductions that call reduce_over_dim or
+ * map_reduce_over_dim for each output element. Automatically
+ * calculates appropriate grain size.
+ */
+template <typename Func>
+[[nodiscard]] bool parallel_for_each_reduce_over_dim_output_index(
```

Comment:
nit: ET_NODISCARD instead

Reply:
why? we have C++17

Reply:
(we should stop using ET_NODISCARD internally, though we probably have to keep it around for backward compatibility)